src/Symfony/Component/AssetMapper/ImportMap/ImportMapAuditor.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\Exception\RuntimeException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapAuditor
+{
+    private const AUDIT_URL = 'https://api.github.com/advisories';
+
+    private readonly HttpClientInterface $httpClient;
+
+    public function __construct(
+        private readonly ImportMapManager $importMapManager,
+        HttpClientInterface $httpClient = null,
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    /**
+     * @return list<ImportMapPackageAudit>
+     */
+    public function audit(): array
+    {
+        $entries = $this->importMapManager->loadImportMapEntries();
+
+        if ([] === $entries) {
+            return [];
+        }
+
+        /** @var array<string, array<string, ImportMapPackageAudit>> $installed */
+        $packageAudits = [];
+
+        /** @var array<string, list<string>> $installed */
+        $installed = [];
+        $affectsQuery = [];
+        foreach ($entries as $entry) {
+            if (null === $entry->url) {
+                continue;
+            }
+            $installed[$entry->importName] ??= [];
+            $installed[$entry->importName][] = $entry->version;
```

Comment:
We can save for another PR, but probably we need to add an optional `packageName` to `ImportMapEntry`. 99% of the time the key (`importName` here) IS the package name. But that's not required. In fact, you can run `importmap:require "vue/dist/vue.esm-bundler.js=vue` to do this exact thing :).

The idea would be that `packageName` never appears in `importmap.php` unless you have changed the key, and then you would need it.